qiskit_experiments/calibration_management/base_calibration_experiment.py

```diff
@@ -0,0 +1,346 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Base class for calibration-type experiments."""
+
+from abc import ABC
+from typing import Any, Dict, Iterable, Optional, Tuple
+
+from qiskit.providers.options import Options
+from qiskit.providers.backend import Backend
+from qiskit.circuit import Parameter
+from qiskit.pulse import ScheduleBlock
+
+from qiskit_experiments.framework.base_experiment import BaseExperiment
+from qiskit_experiments.framework.experiment_data import ExperimentData
+from qiskit_experiments.exceptions import CalibrationError
+
+
+class BaseCalibrationExperiment(BaseExperiment, ABC):
+    """An abstract base class for calibration experiments.
+
+    This abstract base class specifies an experiment and how to update an
+    optional instance of :class:`Calibrations` specified in the calibration options
+    as `calibrations`. Furthermore, the calibration options also specify
+    an auto_update variable which, by default, is set to True. If this variable,
+    is True then the run method of the experiment will call :meth:`block_for_results`
+    and update the calibrations instance once the backend has returned the data.
+
+    Developers that wish to create a calibration experiment must subclass this base
+    class. If the experiment uses custom schedules, which is typically the case, then
+    developers may chose to use the :meth:`get_schedules` method when creating the
+    circuits for the experiment. If :meth:`get_schedules` is used then the developer
+    must override at least one of the following methods used by :meth:`get_schedules`
+    to set the schedules:
+
+    #. :meth:`get_schedules_from_options`
+
+    #. :meth:`get_schedules_from_calibrations`
+
+    #. :meth:`get_schedules_from_defaults`
+
+    These methods are called by :meth:`get_schedules`. Furthermore, developers must implement
+    the :meth:`update_calibrations` which is responsible for updating the values of the
+    parameters stored in an instance of :class:`Calibrations`. This may require the developer
+    to set the class variable :code:`__updater__` if he wishes to use the update classes
+    implemented in :mod:`qiskit_experiments.calibration_management.update_library`. In addition
```

Comment:
This should be more clear on what the subclass required for update is and link to the `BaseUpdater` class. It's also not clear to me if a calibration experiment can work without an updater or not? Do all subclasses need to specify an updater?

Reply:
See here: 9a70f1b . I clarified both of your questions in that commit. Also here:
* A cal. experiment can run without an updater. It will then simply not update.
* All classes should specify `__updater__` but not all classes need to override the `update_calibrations` method.